master/internal/trial.go

```diff
@@ -633,7 +633,7 @@ func (t *trial) processAllocated(
 }
 
 func (t *trial) processCompletedWorkload(ctx *actor.Context, msg workload.CompletedMessage) error {
-	if !t.replaying && (msg.ExitedReason == nil || *msg.ExitedReason == workload.UserCanceled) {
+	if !t.replaying && (msg.ExitedReason == nil || *msg.ExitedReason != workload.Errored) {
```

Comment:
non-blocking: maybe introduce a function like "isExitGracefulStop" that expresses the intent of this exit reason check, and then maybe explicitly check if its in a certain set of reasons, so it's less likely to be busted later?

Reply:
Did the explicit check without defining an extra function since it is simply two or statements (i.e. `*msg.ExitedReason == workload.UserCanceled || *msg.ExitedReason == InvalidHP`. I think it's worth enumerating this without the extra function since a function for checking an or statement doesn't seem to be worth the trouble. Let me know if you think differently on this though.